core/src/main/java/jenkins/scm/RunWithSCM.java

```diff
@@ -99,7 +102,8 @@
             public Iterator<User> iterator() {
                 return new AdaptedIterator<String,User>(culpritIds.iterator()) {
                     protected User adapt(String id) {
-                        return User.get(id);
+                        // TODO: Probably it should not auto-create users
```

Comment:
AFAIK the code is correct as it stands. `/asynchPeople` is going to show synthetic users from changelogs, for example.